draft-bar-cfrg-spake2plus.md

```diff
@@ -233,43 +275,40 @@ specific protocol is given in {{?I-D.ietf-mmusic-sdp-uks}}).
 
 Upon completion of this protocol, A and B compute shared secrets Ka, Ke, KcA,
 and KcB as specified in {{keys}}. B MUST send A a key confirmation message cB
-so both parties agree upon these shared secrets. This confirmation message cB
-is computed as a MAC over the received share (pA) using KcB. Specifically, B
-computes cB = MAC(KcB, pA), where MAC is also a secure PRF. After receipt and
-verification of B's confirmation message, A MUST send B a confirmation message
-using a MAC computed equivalently except with the use of pB and KcA. B MUST NOT
-send application data to A until it has received and verified the confirmation
-message. Key confirmation verification requires recomputation of the MAC and
-checking for equality against that which was received.
+so both parties can confirm that they agree upon these shared secrets. After
+receipt and verification of B's confirmation message, A MUST send B an equivalent
+confirmation message. B MUST NOT send application data to A until it has received
+and verified the confirmation message. Key confirmation verification requires
+recomputation of cA or cB and checking for equality against that which was received.
 
 # Key Schedule and Key Confirmation {#keys}
 
 The protocol transcript TT, as defined in {{spake2plus}},
-is unique and secret to A and B. Both parties use TT to
-derive shared symmetric secrets Ke and Ka as Ke || Ka = Hash(TT). The length of each
-key is equal to half of the digest output, e.g., |Ke| = |Ka| = 128 bits for SHA-256.
-If the required key size is less than half the digest output, e.g. when using SHA-512
-to derive two 128-bit keys, the digest output MAY be truncated.
+is unique and secret to A and B. Both parties use TT to derive shared symmetric
+secrets Ke and Ka. The length of each key is equal to half of the digest output,
+e.g., |Ke| = |Ka| = 128 bits for SHA-256. If the required key size is less than
```

Comment:
Hmm, what *is* the required key size in this case? (Namely, what is the maximum amount of truncation allowed here?)

Reply:
That wording is confusing indeed. It should just say that the shared secret `Ke` can be truncated if needed.